base/strings/substring.jl

```diff
@@ -162,10 +139,15 @@ julia> join(reverse(collect(graphemes("ax̂e")))) # reverses graphemes
 "ex̂a"
 ```
 """
-reverse(s::AbstractString) = RevString(s)
-reverse(s::RevString) = s.string
-
-## reverse an index i so that reverse(s)[i] == s[reverseind(s,i)]
+function reverse(s::AbstractString)::String
+    sprint() do io
```

Comment:
If the input is a `String` or substring thereof, can't we do better than `sprint` because we know exactly the size of the buffer that is required?  i.e. we can just allocate `v = StringVector(sizeof(s))`, write into that, and return `String(v)`?   Also, you can just write codeunits directly, rather than converting to/from `Char` via `s[j]`...

Oh, I guess you are keeping `reverse(s::String)` in string.jl .... it seems like that routine could be generalized to work for `SubString{String}` too, by calling `codeunit` rather than converting the input to `Vector{UInt8}`.

Reply:
👍  I'm honestly still just trying to get all the tests to pass 😅